public/js/src/module/offline-app-worker-partial.js

```diff
@@ -12,14 +12,12 @@ self.addEventListener( 'install', event => {
             .then( cache => {
                 console.log( 'Opened cache' );
 
-                return cache.addAll( resources );
+                // To bypass any HTTP caching, always obtain resource from network
+                return cache.addAll( resources.map( resource => new Request( resource, { cache: 'reload' } ) ) );
```

Comment:
Would this happen…
1. The very first time someone visits the application: the browser would not yet have had any opportunity to cache the resources, so there's effectively no change from the old behavior;
2. Whenever `registration.update()` detects a service worker update: bypassing the browser's HTTP cache, which has been filled at this point, avoids the issue described in #259:
    > If at point 3, the browser is serving a HTTP-cached pre-update version of a Service-Worker-cached resource, there will be a problem with incorrect resources - possibly mismatched, that would not self-resolve because the Service-Worker cache does not expire.

:question: (Just trying to make sure I understand!)

Reply:
1. yes, that's what I think
2. yes, exactly

Reply:
Probably out of scope for this PR, but I was confused about where `resources` was coming from, and I see now that it is templated in (along with `version`) by `offline-controller.js`. While (maybe?) unlikely, it's possible the paths being concatenated will produce syntax errors. Generally this sort of templated code generation makes me nervous. I'd like to consider moving away from that and pass these values to the worker with `postMessage`/receive them with a `message` event.

Reply:
I agree, it's definitely not ideal but so far the best we came up with. I think that using postMessage won't trigger an update of the service worker (and therefore the cache) if a version changes.  It's using a byte-wise comparison of the service worker script. https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API#download_install_and_activate